File: zeroinstall/cmd/destroy.py
"""
The B{0install destroy} command-line interface.
"""

# Copyright (C) 2012, Thomas Leonard
# See the README file for details, or visit http://0install.net.

from zeroinstall.cmd import UsageError

syntax = "PET-NAME"

def add_options(parser):
	pass

def handle(config, options, args):
	"""@type args: [str]"""
	if len(args) != 1:
		raise UsageError()

	pet_name = args[0]

	app = config.app_mgr.lookup_app(pet_name)
	app.destroy()

def complete(completion, args, cword):
	"""@type completion: L{zeroinstall.cmd._Completion}
	@type args: [str]
	@type cword: int"""
	if len(args) != 1: return
	completion.expand_apps()

File: ocaml/destroy.ml
(* Copyright (C) 2013, Thomas Leonard
 * See the README file for details, or visit http://0install.net.
 *)

(** The "0install destroy" command *)

open Options
open Support

let handle options flags args =
  Support.Argparse.iter_options flags (function
    | #common_option as o -> Common_options.process_common_option options o
  );
  match args with
  | [name] -> (
      match Zeroinstall.Apps.lookup_app options.config name with
      | None -> Safe_exn.failf "No such application '%s'" name
      | Some app -> Zeroinstall.Apps.destroy options.config app
  )
  | _ -> raise (Support.Argparse.Usage_error 1)
